app/screens/post_options/options/app_bindings_post_option.tsx

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import withObservables from '@nozbe/with-observables';
+import React, {useCallback, useMemo} from 'react';
+import {map} from 'rxjs/operators';
+
+import {postEphemeralCallResponseForPost} from '@actions/remote/apps';
+import OptionItem from '@components/option_item';
+import {Screens} from '@constants';
+import {ChannelModel} from '@database/models/server';
+import {useAppBinding} from '@hooks/apps';
+import {dismissBottomSheet} from '@screens/navigation';
+import {isSystemMessage} from '@utils/post';
+import {preventDoubleTap} from '@utils/tap';
+
+import type PostModel from '@typings/database/models/servers/post';
+
+type Props = {
+    bindings: AppBinding[];
+    post: PostModel;
+    serverUrl: string;
+    teamId: string;
+}
+
+const AppBindingsPostOptions = ({serverUrl, post, teamId, bindings}: Props) => {
+    const onCallResponse = useCallback((callResp: AppCallResponse, message: string) => {
+        postEphemeralCallResponseForPost(serverUrl, callResp, message, post);
+    }, [serverUrl, post]);
+
+    const context = useMemo(() => ({
+        channel_id: post.channelId,
+        team_id: teamId,
+        post_id: post.id,
+        root_id: post.rootId || post.id,
+    }), [post, teamId]);
+
+    const config = useMemo(() => ({
+        onSuccess: onCallResponse,
+        onError: onCallResponse,
+    }), [onCallResponse]);
+
+    const handleBindingSubmit = useAppBinding(context, config);
+
+    const onPress = useCallback(async (binding: AppBinding) => {
+        const submitPromise = handleBindingSubmit(binding);
+        await dismissBottomSheet(Screens.POST_OPTIONS);
+
+        const finish = await submitPromise;
+        await finish();
+    }, [handleBindingSubmit]);
+
+    if (isSystemMessage(post)) {
+        return null;
+    }
+
+    const options = bindings.map((binding) => (
+        <BindingOptionItem
+            key={binding.location}
+            binding={binding}
+            onPress={onPress}
+        />
+    ));
+
+    return <>{options}</>;
+};
+
+const BindingOptionItem = ({binding, onPress}: {binding: AppBinding; onPress: (binding: AppBinding) => void}) => {
+    const handlePress = useCallback(preventDoubleTap(() => {
+        onPress(binding);
+    }), [binding, onPress]);
+
+    return (
+        <OptionItem
+            label={binding.label}
+            icon={binding.icon}
+            action={handlePress}
+            type='default'
+            testID={`post_options.app_binding.option.${binding.location}`}
+        />
+    );
+};
+
+type OwnProps = {
+    post: PostModel;
+    bindings: AppBinding[];
+}
+
+const withTeamId = withObservables(['post'], ({post}: OwnProps) => ({
+    teamId: post.channel.observe().pipe(map((channel: ChannelModel) => channel.teamId)),
```

Comment:
we should split this to it's own file for readability (0/5 on doing it now or in a different PR)

Now here is the important bit, if the channel for some reason has not been loaded this will cause a crash.
To avoid it, you can change it with the following code

```suggestion
    teamId: post.channelId ? observeChannel(database, post.channelId).pipe(
        switchMap((c) => c.teamId),
        distinctUntilChanged(),
    ) : of$('')
```

Reply:
@enahum I fixed the missing channel issue here. LMK what you think